js/modules.js

```diff
@@ -15,52 +15,79 @@ CORE.create_module("email-nav", function (sb) {
 CORE.create_module("email-catalog", function (sb) {
     var emails;
 
+    var openEmail = function(e) {
+        sb.hide();
+        sb.notify({
+            type : "open-email",
+            data : e.currentTarget.getAttribute("data-email-id")
+        });
+    }
+    var selectEmailCheckbox = function(e) {
+        e.stopPropagation();
+        sb.notify({
+            type : "select-email-checkbox",
+            data : e.currentTarget.getAttribute("data-email-id")
+        });
+    }
+    var markSelectedEmailsAsRead = function() {
+        sb.notify({ 
+            type : "mark-selected-emails-as-read"
+        });
+    }
+
+    var fetchEmails = function(data) {
+        emails = data;
+        sb.template(emails);
```

Comment:
`fetchEmails `gets the email data from "storage" and updates our local `emails` variable. I did it this way so it should be easier to use a real database in the future. so `emails = data` is setting the local `emails` variable with the updated data.